### ISSUE TITLE
Create nightly, dev, and stable build channels

### ISSUE THREAD

**Issue by carreter** [bebop/poly#423]
## Describe the desired feature/enhancement
Set up a CI/CD system that is triggered:
- every night to release a nightly build based on `main`
- on merge into `release-dev` to release a dev build
- on merge into `release-stable` to release a stable build

### Is your feature request related to a problem?
Would simplify the release workflow and allow users to choose between different levels of stability/new features tradeoffs.

### Describe the solution you'd like
Ideally this will be done using a new CI/CD system as suggested in #365, which for now means this issue is blocked by #365. 

## Additional context
@TimothyStiles suggested this and I really like this idea!

**Comment by carreter** [comment]
Nightly builds were introduced by #429. We won't be doing `dev` and `stable` build channels for now, as we are too early in the development cycle to see much of an advantage.
